### DISCUSSION TITLE
IMERG dataset error

### DISCUSSION THREAD

**Discussion by MariRiba** [NASA-LIS/LISF#1534] [Q&A] [answered]
Hi 
I am using GPM IMERG dataset version V07B for precipitation forcing. When I am running the LIS I am getting an error:

> [INFO] Reading HDF5 IMERG precipitation data from /scratch2/mari/lisf/IMERG/201601/3B-HHR.MS.MRG.3IMERG.20160102-S000000-E002959.0000.V07B.HDF5
 Error opening IMERG dataset/Grid/precipitationCal
 [WARN] Error reading IMERG file /scratch2/mari/lisf/IMERG/201601/3B-HHR.MS.MRG.3IMERG.20160102-S000000-E002959.0000.V07B.HDF5
[INFO] LIS cycle time: 01/02/2016 00:30:00

How can I correct it ?

the lis.config file is :
[lis.config_imerg.txt](https://github.com/NASA-LIS/LISF/files/15234449/lis.config_imerg.txt)

**Comment by karsenau** [comment]
@MariRiba:

The GPM IMERG reader in LIS should have been updated in the "master" branch back in Aug-2023, with support for the IMERG V7 final files.  

You may want to check which latest version that you have checked out of LISF, and if it is prior to Aug-2023.

**Comment by sudarsan3010** [comment, marked as answer]
Dear @MariRiba,

This issue is caused because the older IMERG versions (before V7) had precipitation variable in the name of 'precipitationCal' (which is a combination of both IR & Microwave), but in the new IMERG versions (V07), the precipitation variable is named as 'precipitation'.

This has been resolved in the latest available version LISF - 7.5.0. If using 7.4.x or anything lesser, then edit the code in the 'read_IMERG.F90' file located at 'lis/metforcing/imerg' as below,

**Orginal command line**: character(len=40) :: dsetname='/Grid/precipitationCal'
**New command line:** character(len=40) :: dsetname='/Grid/precipitation'

![image](https://github.com/NASA-LIS/LISF/assets/53935951/3771168e-c3e3-4efb-bb19-a8c46cbf0e77)

Hope this helps! :)
